Fails compilation on GHC 9.6

`cabal install ShellCheck -w ghc-9.6.1` fails with:

```
[ 2 of 11] Compiling ShellCheck.Regex ( ShellCheck/Regex.hs, dist/build/ShellCheck/Regex.o, dist/build/ShellCheck/Regex.dyn_o )

ShellCheck/Regex.hs:33:17: warning: [-Wsimplifiable-class-constraints]
    • The constraint ‘RegexMaker
                        Regex CompOption ExecOption String’ matches
        instance RegexMaker Regex CompOption ExecOption String
          -- Defined in ‘Text.Regex.TDFA.String’
      This makes type inference for inner bindings fragile;
        either use MonoLocalBinds, or simplify it using the instance
    • In the type signature:
        make :: RegexMaker Regex CompOption ExecOption String =>
                String -> Regex
      In the expression:
        let
          make ::
            RegexMaker Regex CompOption ExecOption String => String -> Regex
          make = makeRegex
        in make str
      In an equation for ‘mkRegex’:
          mkRegex str
            = let
                make ::
                  RegexMaker Regex CompOption ExecOption String => String -> Regex
                make = makeRegex
              in make str
   |
33 |     let make :: RegexMaker Regex CompOption ExecOption String => String -> Regex
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 3 of 11] Compiling ShellCheck.AST   ( ShellCheck/AST.hs, dist/build/ShellCheck/AST.o, dist/build/ShellCheck/AST.dyn_o )
[ 4 of 11] Compiling ShellCheck.Interface ( ShellCheck/Interface.hs, dist/build/ShellCheck/Interface.o, dist/build/ShellCheck/Interface.dyn_o )
[ 5 of 11] Compiling ShellCheck.Data  ( ShellCheck/Data.hs, dist/build/ShellCheck/Data.o, dist/build/ShellCheck/Data.dyn_o )
[ 6 of 11] Compiling ShellCheck.Formatter.Format ( ShellCheck/Formatter/Format.hs, dist/build/ShellCheck/Formatter/Format.o, dist/build/ShellCheck/Formatter/Format.dyn_o )
[ 7 of 11] Compiling ShellCheck.ASTLib ( ShellCheck/ASTLib.hs, dist/build/ShellCheck/ASTLib.o, dist/build/ShellCheck/ASTLib.dyn_o )
[ 8 of 11] Compiling ShellCheck.Parser ( ShellCheck/Parser.hs, dist/build/ShellCheck/Parser.o, dist/build/ShellCheck/Parser.dyn_o )
[ 9 of 11] Compiling ShellCheck.Analytics ( ShellCheck/Analytics.hs, dist/build/ShellCheck/Analytics.o, dist/build/ShellCheck/Analytics.dyn_o )

ShellCheck/Analytics.hs:2698:38: error: [GHC-64725]
    • Data.Map.insertWith' is gone. Use Data.Map.Strict.insertWith.
    • In the expression: Map.insertWith' (const id) name place read
      In the expression:
        (Map.insertWith' (const id) name place read, written)
      In the first argument of ‘modify’, namely
        ‘(\ (read, written)
            -> (Map.insertWith' (const id) name place read, written))’
     |
2698 |         modify (\(read, written) -> (Map.insertWith' (const id) name place read, written))
```

The bounds should reject GHC-9.6 since it is futile to attempt compilation.

--- a/ShellCheck.cabal
+++ b/ShellCheck.cabal
@@ -46,7 +46,7 @@ library
         semigroups
     build-depends:
       -- The lower bounds are based on GHC 7.10.3
-      -- The upper bounds are based on GHC 9.4.3
+      -- The upper bounds are based on GHC 9.6.1
       aeson                >= 1.4.0 && < 2.2,
       array                >= 0.5.1 && < 0.6,
       base                 >= 4.8.0.0 && < 5,
@@ -54,13 +54,13 @@ library
       containers           >= 0.5.6 && < 0.7,
       deepseq              >= 1.4.1 && < 1.5,
       Diff                 >= 0.4.0 && < 0.5,
-      fgl                  >= 5.7.0 && < 5.8.1.0,
+      fgl                  (>= 5.7.0 && < 5.8.1.0) || (>= 5.8.1.1 && < 5.9),
       filepath             >= 1.4.0 && < 1.5,
-      mtl                  >= 2.2.2 && < 2.3,
+      mtl                  >= 2.2.2 && < 2.4,
       parsec               >= 3.1.14 && < 3.2,
       QuickCheck           >= 2.14.2 && < 2.15,
       regex-tdfa           >= 1.2.0 && < 1.4,
-      transformers         >= 0.4.2 && < 0.6,
+      transformers         >= 0.4.2 && < 0.7,
 
       -- getXdgDirectory from 1.2.3.0
       directory            >= 1.2.3 && < 1.4,

